### DISCUSSION TITLE
No result on analyzed DB

### DISCUSSION THREAD

**Discussion by northsuperman** [github/codeql#9965] [Q&A]
Hello everybody,
I'm trying to analyze a DB I have created, but none of the (standard) queries I ran show any results.
I assume that something wrong happened in the creation of the DB.
The codebase: https://github.com/rxwen/gloox
The command: `codeql database create ../db --language=cpp --overwrite --command=./build.sh`
The building file `build.sh`:
```sh
./configure --with-examples=no --with-tests=no
make -j 8
```
Can someone help me?

**Comment by hmakholm** [comment]
You're not showing which command you use for _running_ queries on the database you create -- I hope you know that's a separate step from `codeql database create`.

You can try
```
codeql query run ../db $CODEQL_DIR/cpp/ql/src/Diagnostics/SuccessfullyExtractedFiles.ql
```
(where `$CODEQL_DIR` is your checkout of https://github.com/github/codeql) to double check that the source files you expect to be extracted have actually been extracted.

If you see all of your files there, are there any alerts in particular you think your code ought to trigger? It might be that the code is just fine! :-)

**Comment by hmakholm** [comment]
(By the way, and unrelated to the actual problem you're having: It will generally be more robust and efficient to do your `configure` step _before_ `codeql database create`, and just give `--command='make -j8'` to the latter. This is particularly relevant if the `configure` script uses something like Autoconf which likes to do a lot of dummy test compilations to figure out what works and what doesn't. You don't generally want those test compiles to be part of what you query with CodeQL. We make reasonable efforts to recognize and ignore them in `database create`, but it would be better not to rely on that).
